fastmri_recon/models/functional_models/mcpdnet.py

```diff
@@ -0,0 +1,80 @@
+"""Learned Primal-dual network adapted to MRI for multi-contrast."""
+import tensorflow as tf
+from tensorflow.keras.layers import Input, Lambda, concatenate, Add
+from tensorflow.keras.models import Model
+
+from ..training.compile import default_model_compile
+from ..utils.complex import conv2d_complex
+from ..utils.fastmri_format import tf_fastmri_format
+from ..utils.fourier import mc_tf_adj_op, mc_tf_op
+
+
+def mcpdnet(input_size=(None, None, None, 1), n_filters=32, lr=1e-3, n_primal=5, n_dual=5, n_iter=10, primal_only=False, fastmri=True, activation='relu'):
```

Comment:
I think this is a great first step, but I would be happier if you were to implement the multi contrast pdnet in a subclassed fashion:
you will be able to leverage the CrossDomain function and also better debug the model